File: fix_links.js
const fs = require('fs');
const path = require('path');
const cheerio = require('cheerio');

const directory = '.';

fs.readdirSync(directory).forEach(file => {
    if (file.endsWith('.html')) {
        const filePath = path.join(directory, file);
        const content = fs.readFileSync(filePath, 'utf-8');
        const $ = cheerio.load(content);
        let modified = false;

        $('a[href="#"], a[href=""]').each((_, a_tag) => {
            const $a = $(a_tag);
            const text = $a.text().trim().toLowerCase();
            const html = $a.html().toLowerCase();

            if (text.includes('shop stock') || text.includes('shop') || text.includes('view all') || text.includes('buy') || html.includes('wedding') || html.includes('corporate') || html.includes('parties') || html.includes('holiday')) {
                $a.attr('href', 'stocked_shop_collection.html');
                modified = true;
            } else if (text.includes('custom design') || text.includes('custom order') || html.includes('custom')) {
                $a.attr('href', 'custom_services_landing.html');
                modified = true;
            } else if (text.includes('about') || text.includes('our story') || text.includes('brand story')) {
                $a.attr('href', 'brand_story.html');
                modified = true;
            } else if (text.includes('stories') || text.includes('community') || html.includes('community')) {
                $a.attr('href', 'community_gallery.html');
                modified = true;
            } else if (text.includes('panta props')) {
                $a.attr('href', 'index.html');
                modified = true;
            } else if (html.includes('public')) {
                $a.attr('href', 'https://twitter.com/pantaprops');
                modified = true;
            } else if (html.includes('photo_camera')) {
                $a.attr('href', 'https://www.instagram.com/pantaprops_/_/');
                modified = true;
            } else if (html.includes('mail')) {
                $a.attr('href', 'mailto:hello@pantaprops.com');
                modified = true;
            } else {
                // Default fallback for any unhandled empty links, like the shop by vibe cards
                $a.attr('href', 'stocked_shop_collection.html');
                modified = true;
            }
        });

        // Convert buttons to have onclick events
        $('button').each((_, btn) => {
            const $btn = $(btn);
            const text = $btn.text().trim().toLowerCase();
            const html = $btn.html().toLowerCase();

            if (text.includes('browse stock') || text.includes('shop')) {
                $btn.attr('onclick', "window.location.href='stocked_shop_collection.html'");
                modified = true;
            } else if (text.includes('start custom order') || text.includes('custom')) {
                $btn.attr('onclick', "window.location.href='custom_services_landing.html'");
                modified = true;
            } else if (text.includes('whatsapp') || html.includes('chat')) {
                $btn.attr('onclick', "window.location.href='https://wa.me/?text=Hi%20Panta!'");
                modified = true;
            }
        });

        // Some large div cards act as buttons in index.html, we can make them clickable too.
        $('.hero-card').each((_, card) => {
            const $card = $(card);
            const text = $card.text().toLowerCase();
            if (text.includes('shop the drop')) {
                $card.attr('onclick', "window.location.href='stocked_shop_collection.html'");
                modified = true;
            } else if (text.includes('make it yours')) {
                $card.attr('onclick', "window.location.href='custom_services_landing.html'");
                modified = true;
            }
        });

        if (modified) {
            fs.writeFileSync(filePath, $.html());
            console.log(`Updated more links and buttons in ${file}`);
        }
    }
});

File: fix_links.py
import os
from bs4 import BeautifulSoup

def update_links(directory):
    for filename in os.listdir(directory):
        if filename.endswith(".html"):
            filepath = os.path.join(directory, filename)
            with open(filepath, 'r') as f:
                soup = BeautifulSoup(f.read(), 'html.parser')

            modified = False
            for a_tag in soup.find_all('a', href=True):
                if a_tag['href'] == '#':
                    text = a_tag.get_text(separator=' ', strip=True).lower()
                    classes = a_tag.get('class', [])
                    
                    if 'shop stock' in text or 'shop' in text and not 'shop the drop' in text:
                        if 'shop all' not in text:
                            a_tag['href'] = 'stocked_shop_collection.html'
                            modified = True
                    if 'custom design' in text or 'custom order' in text:
                        a_tag['href'] = 'custom_services_landing.html'
                        modified = True
                    if 'about' in text or 'our story' in text or 'brand story' in text:
                        a_tag['href'] = 'brand_story.html'
                        modified = True
                    if 'stories' in text or 'community' in text:
                        a_tag['href'] = 'community_gallery.html'
                        modified = True
                    if 'panta props' in text:
                        a_tag['href'] = 'index.html'
                        modified = True
                    
                    # Social links
                    if not text:
                        # Check span content
                        spans = a_tag.find_all('span')
                        for span in spans:
                            if 'public' in span.text:
                                a_tag['href'] = 'https://twitter.com/pantaprops'
                                modified = True
                            if 'photo_camera' in span.text:
                                a_tag['href'] = 'https://instagram.com/pantaprops'
                                modified = True
                            if 'mail' in span.text:
                                a_tag['href'] = 'mailto:hello@pantaprops.com'
                                modified = True

            if modified:
                with open(filepath, 'w') as f:
                    f.write(str(soup))
                print(f"Updated links in {filename}")

if __name__ == "__main__":
    update_links('.')
